benchmark/buffers/buffer-bytelength.js

```diff
@@ -0,0 +1,36 @@
+var common = require('../common');
+
+var bench = common.createBenchmark(main, {
+  encoding: ['utf8', 'base64'],
+  len: [4, 16, 64, 256, 1024],
+  n: [1e7]
+});
+
+// 16 chars of each byte length
+var encodings = {
+  'utf8': {
+    0: 'hello brendan!!!', // 1 byte
+    1: 'ΰαβγδεζηθικλμνξο', // 2 bytes
+    2: '挰挱挲挳挴挵挶挷挸挹挺挻挼挽挾挿', // 3 bytes
+    3: '𠜎𠜱𠝹𠱓𠱸𠲖𠳏𠳕𠴕𠵼𠵿𠸎𠸏𠹷𠺝𠺢' // 4 bytes
+  },
+  'base64': {
+    0: 'aGVsbG8gd29ybGQh', // no padding
+    1: 'YW55IGNhcm5hbCBwbGVhc3VyZS4=', // one byte
+    2: 'YWJjZGVmZ2hpamtsbW5eYw==', // two bytes
+    3: 'YXN1cmUu', // nada
+  }
+};
+
+function main(conf) {
+  var n = conf.n | 0;
+  var len = conf.len | 0;
+  var encoding = conf.encoding;
+  var chars = encodings[encoding];
+
+  bench.start();
+  for (var i = 0; i < n; i++) {
+    Buffer.byteLength(chars[n % 4], encoding);
```

Comment:
If you change the utf8 and base64 objects to arrays, you can replace the hard-coded constant with `n % inputs.length`.